homeassistant/components/ecobee/number.py

```diff
@@ -0,0 +1,94 @@
+"""Support for using number with ecobee thermostats."""
+from __future__ import annotations
+
+import logging
+
+from homeassistant.components.number import NumberEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfTime
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, ECOBEE_MODEL_TO_NAME, MANUFACTURER
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the ecobee thermostat number entity."""
+    data = hass.data[DOMAIN]
+    entities = []
+    _LOGGER.debug("Adding min time home ventilators numbers (if present)")
+    for index in range(len(data.ecobee.thermostats)):
+        thermostat = data.ecobee.get_thermostat(index)
+        if thermostat["settings"]["ventilatorType"] != "none":
+            _LOGGER.debug("Adding 1 ventilator's min time home number")
+            entities.append(
+                EcobeeVentilatorMinTime(
+                    data,
+                    index,
+                    "home",
+                    "ventilatorMinOnTimeHome",
+                    data.ecobee.set_ventilator_min_on_time_home,
+                )
+            )
+
+            entities.append(
+                EcobeeVentilatorMinTime(
+                    data,
+                    index,
+                    "away",
+                    "ventilatorMinOnTimeAway",
+                    data.ecobee.set_ventilator_min_on_time_away,
+                )
+            )
+
+    async_add_entities(entities, True)
+
+
+class EcobeeVentilatorMinTime(NumberEntity):
+    """A number class, representing min time  for an ecobee thermostat with ventilator attached."""
+
+    _attr_has_entity_name = True
+
+    def __init__(self, data, thermostat_index, mode, ecobee_setting_key, set_func):
+        """Initialize ecobee ventilator platform."""
+        self.data = data
+        self.thermostat_index = thermostat_index
+        self.thermostat = self.data.ecobee.get_thermostat(self.thermostat_index)
+        self.ecobee_setting_key = ecobee_setting_key
+        self.set_func = set_func
+        self._attr_name = f"Ventilator min time {mode}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self.thermostat["identifier"])},
+            manufacturer=MANUFACTURER,
+            model=ECOBEE_MODEL_TO_NAME.get(self.thermostat["modelNumber"]),
+            name=self.thermostat["name"],
+        )
+        self._attr_unique_id = f'{self.thermostat["identifier"]}_{mode}'
```

Comment:
Seems a bit week to only add the mode to the identifier?

Reply:
Not sure if you think of a better way, 
technically each ecobee device can only have 1 ventilator, and only 2 move (home and away)

Pairing mode with thermostat name (aka device name) makes it unique

Reply:
In that case we can leave it

Reply:
Probably you should add `ventilator` to the unique_id if somewhere there are other modes coming in not seen today to differentiate it.